group.c

```diff
@@ -58,15 +58,20 @@ static void group_remove(struct Group *g)
   FREE(&g);
 }
 
-int mutt_group_context_clear(struct GroupContext **ctx)
+/**
+ * mutt_group_context_clear: Clear groups of GroupContext list
+ * @param head Head of the GroupContext list
+ */
+int mutt_group_context_clear(struct GroupContextHead *head)
 {
-  struct GroupContext *t = NULL;
-  for (; ctx && *ctx; (*ctx) = t)
+  struct GroupContext *np = STAILQ_FIRST(head), *next = NULL;
+  STAILQ_FOREACH_SAFE(np, head, entries, next)
   {
-    group_remove((*ctx)->g);
-    t = (*ctx)->next;
-    FREE(ctx);
+    group_remove(np->g);
+    STAILQ_REMOVE(head, np, GroupContext, entries);
+    FREE(&np);
   }
+  STAILQ_INIT(head);
```

Comment:
You don't need a STAILQ_INIT if you call STAILQ_REMOVE inside the loop. Please note that STAILQ_REMOVE runs in O(n), which is why I was suggesting to follow the approach used in `mutt_list_free`.